pil-analyzer/src/condenser.rs

```diff
@@ -940,16 +940,32 @@ fn try_value_to_expression<T: FieldElement>(value: &Value<'_, T>) -> Result<Expr
         }
         .into(),
         Value::Closure(c) => try_closure_to_expression(c)?,
-        Value::TypeConstructor(c) => {
+        Value::TypeConstructor(enum_, c) => {
             return Err(EvalError::TypeError(format!(
-                "Type constructor as captured value not supported: {c}."
+                "Type constructor as captured value not supported: {}::{c}.",
```

Comment:
Related to other comment about using structs: it would remove this occurence of `::` by calling a method on the struct.

Reply:
You mean the detail in the error string that this is about a "captured" value should move to the caller since this function could also be used for other values?

Reply:
Yes.

Reply:
-> https://github.com/powdr-labs/powdr/pull/1819